homeassistant/components/ovo_energy/sensor.py

```diff
@@ -0,0 +1,254 @@
+        currency = usage.electricity[len(usage.electricity) - 1].cost.currency_unit
+    except aiohttp.ClientError as exception:
+        _LOGGER.warning(exception)
+        raise PlatformNotReady from exception
+
+    async def async_update_data() -> OVODailyUsage:
+        """Fetch data from OVO Energy."""
+        now = datetime.utcnow()
+        async with async_timeout.timeout(10):
+            return await client.get_daily_usage(now.strftime("%Y-%m"))
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="sensor",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=30),
+    )
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+    async_add_entities(
+        [
+            OVOEnergyLastElectricityReading(coordinator, client),
+            OVOEnergyLastGasReading(coordinator, client),
+            OVOEnergyLastElectricityCost(coordinator, client, currency),
+            OVOEnergyLastGasCost(coordinator, client, currency),
+        ],
+        True,
+    )
+
+
+class OVOEnergySensor(OVOEnergyDeviceEntity):
+    """Defines a OVO Energy sensor."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        client: OVOEnergy,
+        key: str,
+        name: str,
+        icon: str,
+        unit_of_measurement: str = "",
+    ) -> None:
+        """Initialize OVO Energy sensor."""
+        self._coordinator = coordinator
+        self._state = None
+        self._attributes = None
+        self._available = False
+        self._unit_of_measurement = unit_of_measurement
+
+        super().__init__(client, key, name, icon)
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return self._key
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self) -> object:
+        """Return the attributes of the sensor."""
+        return self._attributes
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit this state is expressed in."""
+        return self._unit_of_measurement
+
+    @property
+    def should_poll(self):
+        """No need to poll. Coordinator notifies entity of updates."""
+        return False
+
+
+class OVOEnergyLastElectricityReading(OVOEnergySensor):
+    """Defines a OVO Energy last reading sensor."""
+
+    def __init__(self, coordinator: DataUpdateCoordinator, client: OVOEnergy):
+        """Initialize OVO Energy sensor."""
+
+        super().__init__(
+            coordinator,
+            client,
+            f"{DOMAIN}_{client.account_id}_last_electricity_reading",
```

Comment:
You don't need to prefix it with the domain.

```suggestion
            f"{client.account_id}_last_electricity_reading",
```

This can be applied to all other occurrences as well.